src/lib/tzcode/strptime.c

```diff
@@ -0,0 +1,694 @@
+ *    notice, this list of conditions and the following disclaimer
+ *    in the documentation and/or other materials provided with the
+ *    distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY POWERDOG INDUSTRIES ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE POWERDOG INDUSTRIES BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+ * BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+ * OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * The views and conclusions contained in the software and documentation
+ * are those of the authors and should not be interpreted as representing
+ * official policies, either expressed or implied, of Powerdog Industries.
+ */
+
+#include <sys/cdefs.h>
+
+#ifndef lint
+#ifndef NOID
+static char copyright[] __attribute__((unused)) =
+"@(#) Copyright (c) 1994 Powerdog Industries.  All rights reserved.";
+static char sccsid[] __attribute__((unused)) =
+"@(#)strptime.c	0.1 (Powerdog) 94/03/27";
+#endif /* !defined NOID */
+#endif /* not lint */
+
+#include <ctype.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+
+#include "private.h"
+#include "timelocal.h"
+#include "tzcode.h"
+
+enum flags {
+	FLAG_NONE = 1 << 0,
+	FLAG_YEAR = 1 << 1,
+	FLAG_MONTH = 1 << 2,
+	FLAG_YDAY = 1 << 3,
+	FLAG_MDAY = 1 << 4,
+	FLAG_WDAY = 1 << 5,
+	FLAG_EPOCH = 1 << 6,
+	FLAG_NSEC = 1 << 7,
+};
+
+/*
+ * Calculate the week day of the first day of a year. Valid for
+ * the Gregorian calendar, which began Sept 14, 1752 in the UK
+ * and its colonies. Ref:
+ * http://en.wikipedia.org/wiki/Determination_of_the_day_of_the_week
+ */
+
+static int
+first_wday_of(int year)
+{
+	return (((2 * (3 - (year / 100) % 4)) + (year % 100) +
+		 ((year % 100) / 4) + (isleap(year) ? 6 : 0) + 1) % 7);
+}
+
+char *
+tnt_strptime(const char *__restrict buf, const char *__restrict fmt,
+	     struct tnt_tm *__restrict tm)
+{
+	char c;
+	int day_offset = -1, wday_offset;
+	int week_offset;
+	int i, len;
+	int Ealternative, Oalternative;
+	enum flags flags = FLAG_NONE;
+	int century = -1;
+	int year = -1;
+	const char *ptr = fmt;
+
+	static int start_of_month[2][13] = {
+		{ 0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334, 365 },
+		{ 0, 31, 60, 91, 121, 152, 182, 213, 244, 274, 305, 335, 366 }
+	};
+
+	while (*ptr != 0) {
+		c = *ptr++;
+
+		if (c != '%') {
+			if (isspace((u_char)c))
+				while (*buf != 0 && isspace((u_char)*buf))
+					buf++;
+			else if (c != *buf++)
+				return (NULL);
```

Comment:
Please, don't wrap return values in brackets. We never do that in our code.
You've already modified this file, so I see no point in keeping this strange formatting.

Reply:
Hehe, ok!

Reply:
Done:
```diff
--- a/src/lib/tzcode/strptime.c
+++ b/src/lib/tzcode/strptime.c
@@ -110,7 +110,7 @@ tnt_strptime(const char *__restrict buf, const char *__restrict fmt,
 				while (*buf != 0 && isspace((u_char)*buf))
 					buf++;
 			else if (c != *buf++)
-				return (NULL);
+				return NULL;
 			continue;
 		}
 
@@ -121,19 +121,19 @@ tnt_strptime(const char *__restrict buf, const char *__restrict fmt,
 		switch (c) {
 		case '%':
 			if (*buf++ != '%')
-				return (NULL);
+				return NULL;
 			break;
 
 		case '+':
 			buf = tnt_strptime(buf, Locale->date_fmt, tm);
 			if (buf == NULL)
-				return (NULL);
+				return NULL;
 			flags |= FLAG_WDAY | FLAG_MONTH | FLAG_MDAY | FLAG_YEAR;
 			break;
 
 		case 'C':
 			if (!is_digit((u_char)*buf))
-				return (NULL);
+				return NULL;
 
 			/* XXX This will break for 3-digit centuries. */
 			len = 2;
...
@@ -716,5 +716,5 @@ tnt_strptime(const char *__restrict buf, const char *__restrict fmt,
 		}
 	}
 
-	return ((char *)buf);
+	return (char *)buf;
 }

```